docs/serializer.md

```diff
@@ -0,0 +1,185 @@
+
+## \*\_cached functions
+_Gears-Vk_ features *_cached* function variants of various helper functions for ORCA scene loading to ease serializer usage and avoid constant writing of different code paths for both modes of the serializer. For example, to retrieve a 2D texture coordinates buffer for model data (modelAndMeshes), one can use
+```
+auto texCoordsBuffer = gvk::create_2d_texture_coordinates_buffer_cached(
+	serializer, modelAndMeshes, 0, avk::sync::wait_idle()
+);
+```
+instead of the longer version
+```
+avk::buffer texCoordsBuffer;
+if (serializer.mode() == gvk::serializer::mode::serialize) {
+	texCoordsBuffer = gvk::create_2d_texture_coordinates_buffer(
+		modelAndMeshes, 0, avk::sync::wait_idle()
+	);
+}
+serializer.archive_buffer(texCoordsBuffer);
+```
+The \*\_cached functions internaly take case of either serializing or deserialising the necessary data (in above example the content of the returned buffer, i.e. the 2D texture coordinates). All \*\_cached functions are designed to be easily integrated into existing, non-serialized code paths. The following code shows a non-serialized way of loading a ORCA scene using helper functions of _Gears-Vk_ and creating buffers for positions, indices and texture coordinates. The code omits iterating through all materials and just creates buffers for the first mesh of one material.
+```
+// Load an ORCA scene from file
+auto orca = gvk::orca_scene_t::load_from_file(aPathToOrcaScene);
+// Get all the different materials from the whole scene
+auto distinctMaterialsOrca = orca->distinct_material_configs_for_all_models();
+
+// Get mesh indices of the first material
+std::vector<gvk::model_and_mesh_indices> meshIndices = distinctMaterialsOrca.begin()->second;
+int meshIndicesIndex = 0;
+
+std::vector<std::tuple<avk::resource_reference<const gvk::model_t>, std::vector<size_t>>> modelAndMeshes;
+modelAndMeshes = gvk::make_models_and_meshes_selection(orca->model_at_index(meshIndices[meshIndicesIndex].mModelIndex).mLoadedModel, meshIndices[meshIndicesIndex].mMeshIndices);
+
+// Get a buffer containing all positions, and one containing all indices for all submeshes with this material
+auto [positionsBuffer, indicesBuffer] = gvk::create_vertex_and_index_buffers(
+	modelAndMeshes, {}, avk::sync::wait_idle()
+);
+
+// Get a buffer containing all texture coordinates for all submeshes with this material
+auto texCoordsBuffer = gvk::create_2d_texture_coordinates_flipped_buffer(
+	modelAndMeshes, 0, avk::sync::wait_idle()
+);
+```
+Integrating the serializer into above code is easily done by introducing a condition at the beginning, to avoid scene loading from file, and swap some helper functions for their \*\_cached ones:
```

Comment:
```suggestion
Integrating the serializer into above code is easily done by introducing a condition at the beginning, to avoid scene loading from file, and swap some helper functions for their `*_cached` counterparts:
```